src/components/SubMenu/SubMenu.jsx

```diff
@@ -0,0 +1,52 @@
+import React, {
+    useState,
+    useRef,
+    useEffect
+} from 'react';
+
+import './SubMenu.css';
+
+function SubMenu({ children }) {
+
+    const [isOpen, setIsOpen] = useState(false);
+
+    function useOutside(ref) {
+        useEffect(() => {
+            function handleClickOutside(event) {
+                if (ref.current && !ref.current.contains(event.target)) {
+                    setIsOpen(false);
+                }
+            }
+            document.addEventListener('mousedown', handleClickOutside);
+            return () => {
+                document.removeEventListener('mousedown', handleClickOutside);
+            };
+        }, [ref]);
+    }
```

Comment:
you should follow DRY principle - don't repeat yourself.
I see at least 2 identical hooks.

If you want to use hook, you have to move it to separate function and reuse it in different components.
You also still have to solve it without hook, you should learn basics. Use `onBlur`